src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -624,14 +625,55 @@ private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
                 return typeDeclaration.Identifier.ToString() + typeDeclaration.TypeParameterList;
             }
 
-            private (string? loggerField, bool multipleLoggerFields) FindLoggerField(SemanticModel sm, TypeDeclarationSyntax classDec, ITypeSymbol loggerSymbol)
+            private (string? loggerField, bool multipleLoggerFields) FindLoggerField(SemanticModel sm, TypeDeclarationSyntax classDec,
+                Dictionary<SyntaxTree, SemanticModel> semanticModels, ITypeSymbol loggerSymbol)
             {
                 string? loggerField = null;
 
                 INamedTypeSymbol? classType = sm.GetDeclaredSymbol(classDec, _cancellationToken);
 
                 bool onMostDerivedType = true;
 
+#if ROSLYN4_8_OR_GREATER
+                IEnumerable<ClassDeclarationSyntax> primaryConstructors = classType.InstanceConstructors
+                    .SelectMany(ic => ic.DeclaringSyntaxReferences)
+                    .Select(ds => ds.GetSyntax())
+                    .OfType<ClassDeclarationSyntax>();
+
+                foreach (ClassDeclarationSyntax primaryConstructor in primaryConstructors)
+                {
+                    if (primaryConstructor.ParameterList?.Parameters is SeparatedSyntaxList<ParameterSyntax> parameters)
+                    {
+                        // this may be in a syntax tree without LoggerMessage attributes for which we don't yet have a semantic model
+                        if (!semanticModels.TryGetValue(primaryConstructor.SyntaxTree, out SemanticModel? semanticModel))
+                        {
+                            semanticModel = _compilation.GetSemanticModel(primaryConstructor.SyntaxTree);
+                            semanticModels.Add(primaryConstructor.SyntaxTree, semanticModel);
+                        }
+
+                        foreach (ParameterSyntax parameter in parameters)
+                        {
+                            if (semanticModel.GetSymbolInfo(parameter.Type).Symbol is not ITypeSymbol parameterType)
```

Comment:
you don't need to go to the semantic model for this.  the INmaedTypeSymbol for the class can get you the IMethodSymbol for the primary cosntructor

which has the IParameterSyjmbols.

Reply:
Thanks. I'll have a look.

Reply:
That worked nicely. I've changed it now. Thank you.